server/container/guice/mailbox/src/test/java/org/apache/james/modules/mailbox/PreDeletionHooksConfigurationTest.java

```diff
@@ -0,0 +1,107 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.modules.mailbox;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+
+import org.apache.commons.configuration.ConfigurationException;
+import org.apache.commons.configuration.HierarchicalConfiguration;
+import org.apache.commons.configuration.XMLConfiguration;
+import org.junit.jupiter.api.Test;
+
+import com.google.common.collect.ImmutableList;
+
+import nl.jqno.equalsverifier.EqualsVerifier;
+
+class PreDeletionHooksConfigurationTest {
+    @Test
+    void shouldMatchBeanContract() {
+        EqualsVerifier.forClass(PreDeletionHooksConfiguration.class)
+            .verify();
+    }
+
+    @Test
+    void fromShouldReturnNoneWhenEmpty() throws Exception {
+        HierarchicalConfiguration configuration = new HierarchicalConfiguration();
+
+        assertThat(PreDeletionHooksConfiguration.from(configuration))
+            .isEqualTo(PreDeletionHooksConfiguration.none());
+    }
+
+    @Test
+    void fromShouldThrowWhenInvalidHookConfiguration() throws Exception {
+        XMLConfiguration configuration = new XMLConfiguration();
```

Comment:
Usually we are using _PropertiesConfiguration_.

Reply:
Well, `listeners.xml` is in XML

Also, properties are bad at expressing lists of complex elements.

Finally, keeping easy evolution toward a configuration per-hook seems to be an important concern.

All of this tends to promote XML backed hierarchicalConfiguration.